lib/cc/engine/duplication.rb

```diff
@@ -31,13 +31,16 @@ def run
         languages_to_analyze.each do |language|
           engine = LANGUAGES[language].new(engine_config: engine_config)
           reporter = CC::Engine::Analyzers::Reporter.new(engine_config, engine, io)
-          reporter.run
+
+          Dir.chdir(directory) do
```

Comment:
Would it work to wrap the entire Duplication#run body in the `chdir`? That is closer to what we do in other repos and also closer to the original intention of doing it in the constructor